Spanner/src/Result.php

```diff
@@ -176,25 +182,14 @@ public function rows($format = self::RETURN_ASSOCIATIVE)
     {
         $bufferedResults = [];
         $call = $this->call;
-        $generator = null;
         $shouldRetry = false;
         $isResultsYielded = false;
-        $backoff = new ExponentialBackoff($this->retries, function ($ex) {
-            if ($ex instanceof ServiceException) {
-                return $ex->getCode() === Grpc\STATUS_UNAVAILABLE;
-            }
 
-            return false;
-        });
-
-        $valid = $backoff->execute(function () use ($call, &$generator) {
-            $generator = $call();
-            return $generator->valid();
-        });
+        $valid = $this->createGenerator();
 
         while ($valid) {
```

Comment:
@bshaffer Do we need to change this to `$this->generator->valid()` for more accuracy?